docs/kubernetes/routegroups.md

```diff
@@ -0,0 +1,628 @@
+
+## Mapping Skipper Ingress extensions to RouteGroups
+
+Skipper accepts a set of annotations in Ingress objects that give access to certain Skipper features that would
+not be possible with the native fields of the Ingress spec, e.g. improved path handling or rate limiting. These
+annotations can be expressed now natively in the RouteGroups.
+
+### zalando.org/backend-weights
+
+Backend weights are now part of the backend references, and they can be controlled for multiple backend sets
+within the same route group. See [Gradual traffic switching](#gradual-traffic-switching).
+
+### zalando.org/skipper-filter and zalando.org/skipper-predicate
+
+Filters and predicates are now part of the route objects, and different set of filters or predicates can be set
+for different routes.
+
+### zalando.org/skipper-routes
+
+"Custom routes" in a route group are unnecessary, because every route can be configured with predicates, filters
+and backends without limitations. E.g where an ingress annotation's metadata may look like this:
+
+```yaml
+apiVersion: extensions/v1beta1
+kind: Ingress
+metadata:
+  name: my-ingress
+  zalando.org/skipper-routes: |
+    Method("OPTIONS") -> status(200) -> <shunt>
+spec:
+  backend:
+    serviceName: my-service
+    servicePort: 80
+```
+
+the equivalent RouteGroup would look like this:
+
+```yaml
+apiVersion: zalando.org/v1
+kind: RouteGroup
+metadata:
+  name: my-route-group
+spec:
+  backends:
+  - name: my-backend
+    type: service
+    serviceName: my-service
+    servicePort: 80
+  - name: options200
+    type: shunt
+  defaultBackends:
+  - backendName: my-service
+  routes:
+  - pathSubtree: /
+  - pathSubtree: /
+    methods: OPTIONS
+    filters:
+    - status(200)
+    backends:
+    - options200
+```
+
+### zalando.org/ratelimit
+
+The ratelimiting can be defined on the route level among the filters, in the same format as in this annotation.
+
+### zalando.org/skipper-ingress-redirect and zalando.org/skipper-ingress-redirect-code
+
+Skipper ingress provides global HTTPS redirect, but it allows individual ingresses to override the global
+settings: enabling/disabling it and changing the default redirect code. With route groups, this override can be
+achieved by simply defining an addtional route, with the same matching rules, and therefore the override can be
+controlled eventually on a route basis. E.g:
+
+```yaml
+apiVersion: zalando.org/v1
+kind: RouteGroup
+metadata:
+  name: my-route-group
+spec:
+  backends:
+  - name: my-backend
+    type: service
+    serviceName: my-service
+    servicePort: 80
+  - name: redirectShunt
+    type: shunt
+  defaultBackends:
+  - backendName: my-service
+  routes:
+  - pathSubtree: /
+  - pathSubtree: /
+    predicates:
+    - Header("X-Forwarded-Proto", "http")
+    filters:
+    - redirectTo(302, "https:")
+    backends:
+    - redirectShunt
+```
+
+### zalando.org/skipper-loadbalancer
```

Comment:
Maybe link to https://opensource.zalando.com/skipper/reference/backends/#load-balancer-backend

Reply:
good point, done